### PR TITLE
[SPARK-24537][R]Add array_remove / array_zip / map_from_arrays / array_distinct

### PR DESCRIPTION
**Pull request by huaxingao**
## What changes were proposed in this pull request?
Add array_remove / array_zip / map_from_arrays / array_distinct functions in SparkR.

## How was this patch tested?
Add tests in test_sparkSQL.R

### CODE REVIEW

**Comment by huaxingao** [comment]
@HyukjinKwon @felixcheung 
Could you please review the changes? Thank you very much in advance!

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -3120,6 +3147,24 @@ setMethod("arrays_overlap",
...
+#'
+#' @rdname column_collection_functions
+#' @aliases arrays_zip arrays_zip,Column-method
+#' @note arrays_zip since 2.4.0
+setMethod("arrays_zip",
+          signature(x = "Column"),
+          function(x, ...) {
+            jcols <- lapply(list(x, ...), function(x) {
+              stopifnot(class(x) == "Column")
+              x@jc
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: nit: probably clearer to name the x in function(x) as something else

File: R/pkg/R/functions.R [hunk 2]
```diff
@@ -3120,6 +3147,24 @@ setMethod("arrays_overlap",
...
+#' @details
+#' \code{arrays_zip}: Returns a merged array of structs in which the N-th struct contains all N-th
+#' values of input arrays.
+#'
+#' @rdname column_collection_functions
+#' @aliases arrays_zip arrays_zip,Column-method
+#' @note arrays_zip since 2.4.0
+setMethod("arrays_zip",
+          signature(x = "Column"),
+          function(x, ...) {
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: should add in doc that `...` are Column for this function
https://github.com/huaxingao/spark/blob/87100ec766acf647e91ab14450f1c33c0f929d52/R/pkg/R/functions.R#L198

File: R/pkg/R/functions.R [hunk 3]
```diff
@@ -3071,6 +3085,19 @@ setMethod("array_position",
...
           })
 
+#' @details
+#' \code{array_remove}: Removes all elements that equal to element from the given array.
+#'
+#' @rdname column_collection_functions
+#' @aliases array_remove array_remove,Column-method
+#' @note array_remove since 2.4.0
+setMethod("array_remove",
+          signature(x = "Column", value = "ANY"),
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: should add in doc that for this function, `value` is ...?
https://github.com/huaxingao/spark/blob/87100ec766acf647e91ab14450f1c33c0f929d52/R/pkg/R/functions.R#L193

**Review by felixcheung — COMMENTED** [review]
minor comments, thanks!

**Review by felixcheung — APPROVED** [review]
LGTM

**Review by HyukjinKwon — APPROVED** [review]
LGTM too

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**